test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_work_groups.cpp

```diff
@@ -0,0 +1,291 @@
+            if (global_work_size / local_work_size != resultData[i])
+            {
+                log_error("Data failed to verify: global_work_size != "
+                          "resultData[%zu]=%d\n",
+                          i, resultData[i]);
+                return TEST_FAIL;
+            }
+        // Test Case 1: WGu = WG0 if the user explicitly sets the number of
+        // workgroups (by specifying a non-null local work size) to be WG0 when
+        // the ND-range kernel command is recorded, the new number of workgroups
+        // - if updated - WGu, will be equal to WG0.
+        error = TestUpdateWorkGroups(global_work_size, stream, resultData);
+        if (error != CL_SUCCESS) return error;
+
+        // Test Case 2: WGu < WG0 if the user explicitly sets the number of
+        // workgroups to be WG0 when the ND-range kernel command is recorded,
+        // the new number of workgroups - if updated - WGu, will be less that
+        // WG0.
+        static_assert(update_global_size != 0,
+                      "update_global_size should not be zero");
+        error = TestUpdateWorkGroups(update_global_size, stream, resultData,
+                                     global_work_size);
+        if (error != CL_SUCCESS) return error;
+
+        // Test Case 3: WG0 ≥ WGu > WG1 if the user explicitly sets the number
+        // of workgroups to be WG0 when the ND-range kernel command is recorded,
+        // the new number of workgroups - if updated - WG1, will less that WG0.
+        // Then, call the API function again to update the number of workgroups
+        // to be WGu so that WG0 ≥ WGu > WG1.
+        error = TestUpdateWorkGroups(update_global_size * 2, stream, resultData,
+                                     global_work_size);
+        if (error != CL_SUCCESS) return error;
+
+        return CL_SUCCESS;
+    }
+
+    cl_int TestUpdateWorkGroups(size_t new_global_size, clMemWrapper &stream,
+                                std::vector<cl_int> &resultData,
+                                size_t old_global_size = 0)
+    {
+        cl_int error;
+        cl_mutable_dispatch_config_khr dispatch_config{
+            CL_STRUCTURE_TYPE_MUTABLE_DISPATCH_CONFIG_KHR,
+            nullptr,
+            command,
+            0, // num_args
+            0, // num_svm_arg
+            0, // num_exec_infos
+            0, // work_dim (0 means no change to dimensions)
+            nullptr, // arg_list
+            nullptr, // arg_svm_list (nullptr means no change)
+            nullptr, // exec_info_list
+            nullptr, // global_work_offset
+            &new_global_size, // global_work_size
+            nullptr // local_work_size
+        };
+
+        cl_mutable_base_config_khr mutable_config{
+            CL_STRUCTURE_TYPE_MUTABLE_BASE_CONFIG_KHR, nullptr, 1,
+            &dispatch_config
+        };
+
+        error =
+            clUpdateMutableCommandsKHR(single_command_buffer, &mutable_config);
```

Comment:
`single_command_buffer` needs to be created with [CL_COMMAND_BUFFER_MUTABLE_KHR](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_COMMAND_BUFFER_MUTABLE_KHR) for this to be valid.